### PR TITLE
Align adapter shape with base linear output shape

### PR DESCRIPTION
**Pull request by guyueh1**
> [!IMPORTANT]  
> The `Update branch` button must only be pressed in very rare occassions.
> An outdated branch is never blocking the merge of a PR.
> Please reach out to the automation team before pressing that button.

# What does this PR do ?

Align adapter shape with base linear output shape.

**Collection**: [Note which collection this PR will affect]

# Changelog

- Add specific line by line info of high level changes in this PR.

# Usage

- You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"

**Pre checks**:

- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:

- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed.
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information

- Related to # (issue)

### CODE REVIEW

File: nemo/collections/llm/peft/canonical_lora.py [hunk 1]
```diff
@@ -81,7 +81,7 @@ def forward(self, x):
         qkv_4d = torch.cat([query_4d, key_4d, value_4d], dim=2)
         adapter_output = qkv_4d.reshape(qkv_4d.shape[0], qkv_4d.shape[1], -1)
 
-        return linear_output + adapter_output, bias
+        return linear_output + adapter_output.reshape(linear_output.shape), bias
```
Comment (ananthsub) [inline nemo/collections/llm/peft/canonical_lora.py:84]: is it needed for the FC1 upgate too in L102?

Comment (guyueh1) [reply nemo/collections/llm/peft/canonical_lora.py:84]: My test doesn't cover that so I'm not certain there is a bug, but I added it just because it's no cost if their shapes are already the same

**Comment by chtruong814** [comment]
@guyueh1 that one failure is a known issue unrelated to this PR.  If you can get a reviewer to approve, then I can merge this for you.

**Review by ananthsub — APPROVED** [review]

**Comment by chtruong814** [comment]
Moving forward with merging this.  Previous CI run had a known test failure unrelated to this change.
